src/network.hpp

```diff
@@ -0,0 +1,119 @@
+#pragma once
+
+// opencv
+#include <opencv2/core.hpp>
+
+// spdlog
+#include <spdlog/spdlog.h>
+
+// std
+#include <algorithm>
+#include <random>
+#include <vector>
+
+// tuple holding information about input of a training data and expected output
+// out of it
+struct Sample {
+public:
+  Sample() = default;
+  Sample(const std::vector<double> &input,
+         const std::vector<double> &expected_output)
+      : input(input), expectedOutput(expected_output) {}
+
+public:
```

Comment:
In structs everything is public by default so no need to add the public keywords.